roles/ood/tasks/main.yaml

```diff
@@ -47,6 +47,29 @@
     src: cluster.yml
     dest: /etc/ood/config/clusters.d/{{ cluster_name }}.yml
 
+- name: Creates directory to clone the jupyter app
+  file:
+    path: /var/www/ood/apps/sys/jupyter/
+    state: directory
+
+- name: Clone the default jupyter app from OOD
+  git:
+    repo: https://github.com/OSC/bc_osc_jupyter.git 
+    version: master
+    dest: /var/www/ood/apps/sys/jupyter
+
+- name: Add cluster name for the jupyter app to use
+  lineinfile:
+    path: /var/www/ood/apps/sys/jupyter/form.yml
+    regexp: '^cluster:'
+    line: 'cluster: {{ cluster_name }}'
+
+- name: Remove content from submit.yml.erb
+  command: /usr/bin/sed -ie '/^/,$d' /var/www/ood/apps/sys/jupyter/submit.yml.erb
+ 
+- name: use the basic web server template for generating the job script
```

Comment:
Could you explain why this replacement of the existing template is needed?

Also there is likely a cleaner way to accomplish this than erasing the content in one task and printing the content in the next.  Consider a template for this file.

Reply:
The default template would throw an error "no implicit conversion of Hash into Array" when starting the up the app in OOD portal. I think it is custom templating used by the OSC folks, so I replaced it with the template in basic Jupyter example here.(https://github.com/OSC/bc_example_jupyter/blob/master/submit.yml.erb)

Will implement the template.

Reply:
An erb file is a Ruby language template.  This is typical of a puppet or chef provisioning environment.  It might be good to have a work-alike for ansible.

Reply:
And come to think of it, since OOD is written in Ruby is this template markup signficant inside ood?  That is, does this need to remain an erb  template so it can be properly consumed by ood?

Reply:
This issue has been discussed in https://github.com/OSC/bc_osc_jupyter/issues/11

It looks like they  have moved their SLURM support to https://github.com/OSC/bc_example_jupyter/

Reply:
Just did a simple test and turns out, that error is cause by key "resources" inside "native". Seems like "resources" is not supported by slurm. The ruby part is totally fine to stay in the erb file. We can keep it for the same purpose, create option for gpu node requests.